src/v2_services/chat_service.py

```
@@ -411,9 +434,74 @@ async def generate_answer(
+        save_to_history(session_id, search_query, final_answer)
+
+    # ----- Step 7: 결과 반환 -----
     return {
         "answer": final_answer,
         "related_law_id_list": law_ids,
         "search_success": True,
     }
+
+
+async def generate_answer_stream(
+    query: str, db: AsyncSession, country_id: int, session_id: str = None
```

Ruff (0.15.4)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)